src/config.ts

```diff
@@ -60,32 +66,63 @@ export default class ConfigService {
       ? await loadViaHttp(this.#uri)
       : require(/* webpackIgnore: true */ this.#uri).default;
 
-    const [renderConfig, errors] = validateRenderConfig(config);
+    const [validConfig, errors] = validateConfig(config);
 
     if (errors !== undefined) {
       throw new Error(errors?.message);
     }
 
-    logger.info(`Render config valid. ${renderConfig.size} styles available.`);
+    logger.info(
+      `Render config valid. ${validConfig.renderConfig.size} styles available.`
+    );
 
-    this.#config = renderConfig;
+    this.#currentVersion = validConfig.version;
+    this.#renderConfig = validConfig.renderConfig;
   }
 
-  setConfig(style: string, config: RenderDescriptor) {
-    this.#config.set(style, config);
+  /**
+   * Set a specific configuration for rendering. Primarily used for testing.
+   *
+   * @internal
+   */
+  setRenderConfig(style: string, config: RenderDescriptor) {
+    this.#renderConfig.set(style, config);
+  }
+
+  /**
+   * Set the current config version. Primarily used for testing.
+   *
+   * @internal
+   */
+  setVersion(version: string) {
+    this.#currentVersion = version;
   }
 
   /**
    * Get a render style configuration given the style key
    */
   getConfig(style: string) {
-    return this.#config.get(style);
+    return this.#renderConfig.get(style);
   }
 
   /**
    * Get the list of available render style configurations
    */
-  renderStyles() {
-    return this.#config.keys();
+  get renderStyles() {
+    return this.#renderConfig.keys();
+  }
+
+  /**
+   * Checks that we have loaded a configuration file
+   */
+  get isLoaded() {
+    return this.#currentVersion !== null;
```

Comment:
I'm not sure if tying the `isLoaded` state to a version being set. My spidey senses say we're gonna eventually regret this in the future but cannot put my finger on it.

Reply:
I think if we run into problems with this it should be something we can change without much problem.

Right now I see this service as being "once configured you cannot unconfigure it".

Maybe that will change.